### ISSUE TITLE
File size issue

### ISSUE THREAD

**Issue by Nakkie040** [xibosignage/xibo#412]
Hi All,

We are currently having an issue with the file size in XIBO. 
Our PHP.ini file is updated with new a new post_max_size (1024m) and a new upload_max_size (1024m).
When we try to upload a new large file (250mb+) the upload restarts multiple times without any user input and after that it stops at around 100mb with the notification true.

is there someone who can help me in the right direction ?

thanks !

**Comment by alexharrington** [comment]
This may be of help:
https://community.xibo.org.uk/t/how-do-i-upload-files-bigger-than-2mb-how-can-i-increase-the-upload-file-size-limit/27

**Comment by nielsvdc** [comment]
Hi Alex,

The problem still exists. These are our php.ini settings:

upload_max_filesize=1024M
post_max_size=1024M
memory_limit=128M
max_execution_time=180
max_input_time=300

Uploading the 250Mb file should take about 1 minute according to the upload counter. But during the upload the upload stops between 30-40% and then restarts 3 times to retry the upload. After failing the upload the video will display the message "true". Please see the screenshot.

![image](https://cloud.githubusercontent.com/assets/10850600/6596139/a41ecaa0-c7f1-11e4-880b-1cfd4fe4bbb6.png)

**Comment by alexharrington** [comment]
Please check the FAQ again - specifically the link to the PHP documentation
on memory_limit

memory_limit must be set at least as large as post_max_size or generally it
should be set larger (and you'll need to have that much RAM available in
the server to handle that too).

You might also want to check the server library temp directory isn't full
of half uploaded files. If it is, you can safely delete those.

**Comment by nielsvdc** [comment]
I set the memory_limit to 2G (server has 16G), but still no solution. No files in temp dir, disk has 3TB space left.

**Comment by alexharrington** [comment]
Is the library directory definitely writeable then as you webserver's user?

**Comment by nielsvdc** [comment]
Yes. We are able to upload smaller files.

**Comment by alexharrington** [comment]
Try adjusting max_input_time

Also ensure you're restarting your webserver service after each
modification of the php.ini file (otherwise they won't have any effect).

Also be sure any intermediate proxy server isn't limiting the size of files
being uploaded.

**Comment by nielsvdc** [comment]
Unfortunately still not working. We have an external hosted server and just found out it's running nginx instead of apache. But this should not be a problem. Changed all php.ini settings, including max_input_time to high values and even restarted the server, instead of only restarting httpd and nxginx services. Still no luck on uploading large video files.

**Comment by alexharrington** [comment]
That makes a big difference. If you look at the FAQ over on the community
site you need to ensure you've configured nginx appropriately too

**Comment by nielsvdc** [comment]
In my defence, I didn't hear of nginx until this morning. So a new world opened up today :-). Read the article you send on your first reaction, changed the nginx settings and it works. But now running into a "internal server error" after upload finished.

**Comment by alexharrington** [comment]
What's in the log?

**Comment by nielsvdc** [comment]
please check you mail

**Comment by alexharrington** [comment]
Looks like some of your modules aren't installed correctly. Please can you
run that from the Modules page in the CMS.

Other than that you may need to enable debug and audit and put the server
in to test mode to get more detailed logging

**Comment by nielsvdc** [comment]
Upgraded from 1.7.1 to 1.7.2 but still those error. Verifying modules doesn't help.

Put server into test mode, but no php errors on uploading the file. Very strange.

**Comment by alexharrington** [comment]
You need test mode, debug and audit. That will give an indication of where
in the upload process it's failing. Just because there's no error in the
log doesn't mean it's not useful.

Please run through the report fault wizard as that's the easiest way to be
sure to have the correct setup.

**Comment by nielsvdc** [comment]
OK, so this was quite an adventure through the world of Plesk, apache and nginx.
Finally I found the setting that was not changed. I found in the http error_log the error message "mod_fcgid: HTTP request length 134223251 (so far) exceeds MaxRequestLen (134217728)". So along with php settings and nginx settings, there are also mod_fcgid settings in de last_httpd.conf file. The settings FcgidMaxRequestLen also be changed to the same upload value used for php and nginx. This fixed the internal server error message.
For reference, this article lead me to the solution http://goo.gl/Q0mdu9.

I also fixed the module error message by copying the missing js files to the location where they were missing.

Thanks for your patience Alex.
